### DISCUSSION TITLE
multiple reverse-proxy instances in a single mitmproxy?

### DISCUSSION THREAD

**Discussion by gabor** [mitmproxy/mitmproxy#5373] [General]
hi,

i have a use-case, which i can solve by running 4 copies of mitmproxy. they all run in reverse-proxy mode, with different hosts&ports. something like:
```
mitmproxy --mode-reverse:http://localhost:2000 --listen-port=4000
mitmproxy --mode-reverse:http://localhost:2001 --listen-port=4001
mitmproxy --mode-reverse:http://localhost:2002 --listen-port=4002
mitmproxy --mode-reverse:http://localhost:2003 --listen-port=4003
```

this works fine, but i have 4 user-interfaces that i have to switch between. is there a way to achieve this with a single mitmproxy process? (or having multiple mitmproxies render into the same UI somehow?)

**Comment by mhils** [comment]
See https://github.com/mitmproxy/mitmproxy/discussions/5288 :-)
